xapian-core/cluster/cluster.cc

```diff
@@ -0,0 +1,391 @@
+    termcount size;
+    bool started;
+  public:
+    PointTermIterator(const unordered_map<string, double> &termlist) :
+    i(termlist.begin()), end(termlist.end()), size(termlist.size()), started(false)
+    {}
+    termcount get_approx_size() const { return size; }
+    termcount get_wdf() const { throw UnimplementedError("PointIterator doesn't support get_wdf()"); }
+    string get_termname() const { return i->first; }
+    doccount get_termfreq() const { throw UnimplementedError("PointIterator doesn't support get_termfreq()"); }
+    Internal * next();
+    termcount positionlist_count() const {
+	 throw UnimplementedError("PointTermIterator doesn't support positionlist_count()");
+    }
+    bool at_end() const;
+    PositionIterator positionlist_begin() const {
+	throw UnimplementedError("PointTermIterator doesn't support positionlist_begin()");
+    }
+    Internal * skip_to(const string &term);
+};
+
+TermIterator::Internal *
+PointTermIterator::next() {
+    if (!started) {
+	started = true;
+	return NULL;
+    }
+    Assert(i != end);
+    ++i; return NULL;
+}
+
+bool
+PointTermIterator::at_end() const
+{
+    if (!started) return false;
+    return i == end;
+}
+
+TermIterator::Internal *
+PointTermIterator::skip_to(const string &term) {
+    if (i->first == term)
+	return NULL;
+    while (i->first != term)
+	i++;
+    return NULL;
+}
+
+TermIterator
+PointType::termlist_begin() const {
+    LOGCALL(API, TermIterator, "PointType::termlist_begin()", NO_ARGS);
+    return TermIterator(new PointTermIterator(values));
+}
+
+TermIterator
+PointType::termlist_end() const {
+    LOGCALL(API, TermIterator, "PointType::termlist_end()", NO_ARGS);
+    return TermIterator(NULL);
+}
+
+bool
+PointType::contains(string term) {
+    LOGCALL(API, bool, "PointType::contains()", term);
+    return !(values.find(term) == values.end());
+}
+
+double
+PointType::get_value(string term) {
+    LOGCALL(API, double, "Point::get_value()", term);
+    return (values.find(term) == values.end()) ? 0.0 : values[term];
+}
+
+double
+PointType::get_magnitude() const {
+    LOGCALL(API, double, "PointType::get_magnitude()", NO_ARGS);
+    return magnitude;
+}
+
+void
+PointType::add_value(string term, double value) {
+    LOGCALL_VOID(API, "PointType::add_value()", term | value);
+    unordered_map<string, double>::iterator it;
+    it = values.find(term);
+    if (it != values.end())
+	it->second += value;
+    else
+	values[term] = value;
+}
+
+void
+PointType::set_value(string term, double value) {
+    LOGCALL_VOID(API, "PointType::set_value()", term | value);
+    values[term] = value;
+}
+
+int
+PointType::termlist_size() const {
+    LOGCALL(API, int, "PointType::termlist_size()", NO_ARGS);
+    return values.size();
+}
+
+Document
+Point::get_document() const {
+    LOGCALL(API, Document, "Point::get_document()", NO_ARGS);
+    return doc;
+}
+
+void
+Point::initialize(TermListGroup &tlg, const Document &doc_) {
```

Comment:
That was because of using the `[ ]` operator within the get_termfreq(), which was not letting the TermListGroup be const. I've replaced that by the `at()` function, which lets the object be const, so that we can pass it as const reference.